client/src/search.tsx

```diff
@@ -333,7 +301,11 @@ function InnerSearchNavigateWidget(props: InnerSearchNavigateWidgetProps) {
           onKeyDown: (event) => {
             if (event.key === "Escape" && inputRef.current) {
               inputRef.current.blur();
-            } else if (event.key === "Enter" && inputValue.trim()) {
+            } else if (
+              event.key === "Enter" &&
+              inputValue.trim() &&
+              highlightedIndex === -1
```

Comment:
Whether "Enter" was taking me to a page or the search page for that query felt as flaky as a coin flip and I think what we'd actually want is to only navigate when nothing is selected. Hence the check for what's highlighted (which should represent what's selected)